backend/app/core/auth.py

```
@@ -54,84 +69,103 @@ def generate_random_token(length: int = 32) -> str:
-    """Verify and decode a token."""
+    """Verify and decode a token with proper error handling."""
     try:
         payload = jwt.decode(
             token, settings.secret_key, algorithms=[settings.algorithm]
         )
         if payload.get("type") != expected_type:
-            return None
+            raise TokenTypeError(f"Expected {expected_type} token")
         return payload
+    except ExpiredSignatureError:
+        raise TokenExpiredError()
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)